### ISSUE TITLE
ARG002 and Protocols

### ISSUE THREAD

**Issue by nstarman** [charliermarsh/ruff#1741]
ARG002 is raised for `typing.Protocol` classes, which generally do not have an implementation.
I don't think these should be reported, or at suppressed by a configuration option.

```python

class Example(Protocol):
    def method(self, arg1: object) -> object:
        ...
```

**Comment by charliermarsh** [comment]
The above example actually doesn't trigger `ARG002` for me:

```py
class Example(Protocol):
    def method(self, arg1: object) -> object:
        ...
```

```
❯ ruff --isolated foo.py --select ARG -n
```

(We omit methods with empty bodies.)

It _does_ trigger if I include a docstring, which is probably incorrect:

```py
class Example(Protocol):
    def method(self, arg1: object) -> object:
        """Some docstring."""
        ...
```

**Comment by charliermarsh** [comment]
If that _is_ the issue, it's fixed in #1742.

**Comment by charliermarsh** [comment]
(Just LMK if that's _not_ the issue and I'll re-open!)

**Comment by nstarman** [comment]
That does appear to be the issue, thanks! I have docstrings in my Protocols.
